Simplenote/AuthViewController+Swift.swift

```diff
@@ -62,21 +61,18 @@ extension AuthViewController {
     }
 
     func refreshButtonTitles() {
-        let actionText    = signingIn ? Localization.signInAction   : Localization.signUpAction
-        let tipText       = signingIn ? Localization.signUpTip      : Localization.signInTip
-        let switchText    = signingIn ? Localization.signUpAction   : Localization.signInAction
-
-        actionButton.title         = actionText
-        switchTipField.stringValue = tipText.uppercased()
-        switchActionButton.title   = switchText.uppercased()
+        actionButton.title          = mode.primaryActionText
+        secondaryActionButton.title = mode.secondaryActionText?.uppercased() ?? ""
+        switchTipField.stringValue  = mode.switchActionTip.uppercased()
+        switchActionButton.title    = mode.switchActionText.uppercased()
     }
 
     /// Makes sure unused components (in the current mode) are effectively disabled
     ///
     func refreshEnabledComponents() {
-        passwordField.isEnabled = signingIn
-        forgotPasswordButton.isEnabled = signingIn
-        wordPressSSOButton.isEnabled = signingIn
+        passwordField.isEnabled         = mode.isPasswordVisible
+        secondaryActionButton.isEnabled = mode.isSecondaryActionVisible
+        wordPressSSOButton.isEnabled    = mode.isWordPressVisible
```

Comment:
Secondary Action refers to either `Forgot Password` or `Continue with Password` actions